lib/jsdom/living/webstorage/Storage-impl.js

```diff
@@ -0,0 +1,104 @@
+"use strict";
+
+const fs = require("pn/fs");
+const DOMException = require("domexception");
+const StorageEvent = require("../generated/StorageEvent");
+const idlUtils = require("../generated/utils");
+const { convertMapToObject } = require("../../utils");
+
+// https://html.spec.whatwg.org/multipage/webstorage.html#the-storage-interface
+class StorageImpl {
+  constructor(args, privateData) {
+    this._items = new Map();
```

Comment:
What are the advantages of using a map over an object, if you're going to just convert it to an object later?

Reply:
I originally used a Map before I saw that there could be a need for stringification, knowing that (at least historically) the property order has not been entirely [reliable](https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order).

I did try an object and found while they performed about the same, the code became less clean/consistent (given `Object.keys(x)` vs `x.size` / `delete x[y]` vs `x.delete(y)` with the object, so I've kept the Map. Now that I've removed the writeToDisk-ability, stringification is not as important and I've opted for string concatenation of the keys + values to determine the storage size instead.